From d4e74f9f1d5379cb1a9824d6df9f8b0602adfd2d Mon Sep 17 00:00:00 2001
From: Chen Jiahao <chenjiahao16@huawei.com>
Date: Sat, 29 Jan 2022 00:22:53 +0800
Subject: [PATCH] KABI: reserve space for struct input_dev
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: d4e74f9f1d5379cb1a9824d6df9f8b0602adfd2d
Modified-by-SEL: No


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4QWCF
CVE: NA

--------

Reserving some fields in struct input_dev for future development.

Signed-off-by: Chen Jiahao <chenjiahao16@huawei.com>
Reviewed-by: Liao Chang <liaochang1@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/input.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/input.h b/include/linux/input.h
index 56f2fd32e609..08d1409ce222 100644
--- a/include/linux/input.h
+++ b/include/linux/input.h
@@ -20,6 +20,7 @@
 #include <linux/fs.h>
 #include <linux/timer.h>
 #include <linux/mod_devicetable.h>
+#include <linux/kabi.h>
 
 struct input_dev_poller;
 
@@ -201,6 +202,9 @@ struct input_dev {
 	bool devres_managed;
 
 	ktime_t timestamp[INPUT_CLK_MAX];
+
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
 };
 #define to_input_dev(d) container_of(d, struct input_dev, dev)
 
-- 
2.26.2

